okhttp/src/main/java/okhttp3/CertificatePinner.kt

```diff
@@ -305,23 +310,16 @@ class CertificatePinner internal constructor(
         publicKey.encoded.toByteString().sha256()
 
     internal fun newPin(pattern: String, pin: String): Pin {
-      val canonicalHostname = when {
-        pattern.startsWith(WILDCARD) -> {
-          "http://${pattern.substring(WILDCARD.length)}".toHttpUrl().host
-        }
-        else -> {
-          "http://$pattern".toHttpUrl().host
-        }
-      }
+      val canonicalPattern = "https://$pattern".toHttpUrl().host
```

Comment:
This seems questionable since * isn't a valid host character. Why simplify here?

Reply:
Yes, but it does seem to work fine. This is required for punycode expansion, otherwise, existing tests would fail.

Reply:
Yes, but maybe '(?:\\*{1,2}\\.)+' should be stripped before converting in case we make HttpUrl stricter here?

Reply:
Now using String.toCanonicalHost() directly